devito/types/caching.py

```diff
@@ -168,9 +168,13 @@ def clear(cls, force=True):
 
         # Wipe out the hidden module-private SymPy caches
         sympy.polys.rootoftools.ComplexRootOf.clear_cache()
-        sympy.polys.rings._ring_cache.clear()
-        sympy.polys.fields._field_cache.clear()
         sympy.polys.domains.modularinteger._modular_integer_cache.clear()
+        try:
+            sympy.polys.rings._ring_cache.clear()
+            sympy.polys.fields._field_cache.clear()
+        except AttributeError:
+            # SymPy 1.14 and later
```

Comment:
since I'm paranoid, could you try a `git grep _cache` in SymPy 1.14 to be sure they haven't added any other random caches?